tools/calyx-pass/src/tui.rs

```diff
@@ -0,0 +1,146 @@
+use console::Term;
+use std::{cmp::min, io::Write};
+
+/// Implements a scrollback buffer.
+///
+/// # Example
+/// ```
+/// fn example(output: &Term) {
+///     let scrollback_buffer = ScrollbackBuffer::new(output);
+///     writeln!(scrollback_buffer, "hi");
+///     scrollback_buffer.display();
+///     scrollback_buffer.scroll_down();
+///     scrollback_buffer.clear();
+/// }
+/// ```
+pub struct ScrollbackBuffer {
```

Comment:
Does this adapt automatically when the terminal window is resized?

Reply:
Not yet, that's the next thing to work on!

Reply:
I believe it should now.